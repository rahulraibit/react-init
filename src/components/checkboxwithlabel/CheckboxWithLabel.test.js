import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CheckboxWithLabel from './CheckboxWithLabel';
import renderer from 'react-test-renderer';

// automatically unmount and cleanup DOM after the test is finished.
// afterEach(cleanup);
test('Test CheckBoxWith label rendrer', () => {
    let component = renderer.create(
        <CheckboxWithLabel labelOn="On"/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <CheckboxWithLabel labelOff="Off"/>
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
})
it('CheckboxWithLabel changes the text after click', () => {
    const { queryByLabelText, getByLabelText } = render(
        <CheckboxWithLabel labelOn="On" labelOff="Off" />,
    );

    expect(queryByLabelText(/off/i)).toBeTruthy();

    fireEvent.click(getByLabelText(/off/i));

    expect(queryByLabelText(/on/i)).toBeTruthy();
});