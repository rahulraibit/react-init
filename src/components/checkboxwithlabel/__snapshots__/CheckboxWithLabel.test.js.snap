// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckBoxWith label rendrer 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;

exports[`Test CheckBoxWith label rendrer 2`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;
